pycbc/events/coherent.py

```diff
@@ -0,0 +1,371 @@
+# Copyright (C) 2022 Andrew Williamson
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+""" This module contains functions for calculating and manipulating coherent
+triggers.
+"""
+
+import numpy as np
+
+
+def get_coinc_indexes(idx_dict, time_delay_idx):
+    """Return the indexes corresponding to coincident triggers
+
+    Parameters
+    ----------
+    idx_dict: dict
+        Dictionary of indexes of triggers above threshold in each
+        detector
+    time_delay_idx: dict
+        Dictionary giving time delay index (time_delay*sample_rate) for
+        each ifo
+
+    Returns
+    -------
+    coinc_idx: list
+        List of indexes for triggers in geocent time that appear in
+        multiple detectors
+    """
+    coinc_list = np.array([], dtype=int)
+    for ifo in idx_dict.keys():
+        # Create list of indexes above threshold in single detector in geocent
+        # time. Can then search for triggers that appear in multiple detectors
+        # later.
+        if len(idx_dict[ifo]) != 0:
+            coinc_list = np.hstack(
+                [coinc_list, idx_dict[ifo] - time_delay_idx[ifo]]
+            )
+    # Search through coinc_idx for repeated indexes. These must have been loud
+    # in at least 2 detectors.
+    coinc_idx = np.unique(coinc_list, return_counts=True)[0][
+        np.unique(coinc_list, return_counts=True)[1] > 1
+    ]
```

Comment:
This may be more readable like this:
```python
something = np.unique(coinc_list, return_counts=True)
coinc_idx = something[0][something[1] > 1]
```

Reply:
Done in f222a2